core/src/main/java/org/elasticsearch/client/Client.java

```diff
@@ -128,7 +142,9 @@
      * <p>
      * The id is optional, if it is not provided, one will be generated automatically.
      */
-    IndexRequestBuilder prepareIndex();
+    default IndexRequestBuilder prepareIndex() {
```

Comment:
We've discussed this previously in #18468 and do not that these should be `default` methods.